File: 3-leetcode/02-medium/js/2825-make-string-a-subsequence-using-cyclic-increments.ts
const canMakeSubsequence = (str1: string, str2: string): boolean => {
  let i = 0;
  let j = 0;
  let n = str1.length;
  let m = str2.length;

  while (i < n && j < m) {
    if ((str2.charCodeAt(j) - str1.charCodeAt(i) + 26) % 26 <= 1) {
      j++;
    }
    i++;
  }

  return j === m;
};

File: 3-leetcode/02-medium/dart/2825-make-string-a-subsequence-using-cyclic-increments.dart
class Solution {
  bool canMakeSubsequence(String str1, String str2) {
    int i = 0;
    int j = 0;
    int n = str1.length;
    int m = str2.length;

    while (i < n && j < m) {
      if ((str2.codeUnitAt(j) - str1.codeUnitAt(i) + 26) % 26 <= 1) {
        j++;
      }

      i++;
    }

    return j == m;
  }
}
